client/src/app/core/services/websocket.service.ts

```diff
@@ -0,0 +1,42 @@
+import { Injectable } from '@angular/core';
+import { Router } from '@angular/router';
+import { webSocket, WebSocketSubject } from 'rxjs/webSocket';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class WebsocketService {
+    constructor(private router: Router) {}
+
+    //might be any for simplicity or MessageEvent or something different
+    private subject: WebSocketSubject<any>;
+
+    public connect(): WebSocketSubject<any> {
+        let socketProtocol = this.getWebSocketProtocoll();
+        let socketPath = this.getWebSocketPath();
+        const socketServer = window.location.hostname + ':' + window.location.port;
+        if (!this.subject) {
+            this.subject = webSocket(socketProtocol + socketServer + socketPath);
+        }
+        return this.subject;
+    }
+
+    // delegates to websockets for either the side or projector websocket
+    private getWebSocketPath(): string {
+        const currentRoute = this.router.url;
+        if (currentRoute.includes('projector')) {
```

Comment:
A little note: Can you make this more specific. We later may have routes like `/manage-projectors/`, where this check should be false. E.g. explicit a startsWith with `/projector/` and `/real-projector/`.